docs/build/tools/foundry.mdx

```diff
@@ -0,0 +1,281 @@
+      out = "out"
+      libs = ["lib", "dependencies"]
+      // highlight-start
+      ffi = true
+      
+      [dependencies]
+      "@oasisprotocol-sapphire-contracts" = "0.2.13"
+      "@oasisprotocol-sapphire-foundry" = "0.1.0"
+      // highlight-end
+   ```
+
+   Note:
+      - `ffi = true` enables `vm.ffi()` ([foreign function interface])
+      which is used to call rust bindings containing the precompile and
+      decryption logic
+      - `"dependencies"` lists the required project dependencies installed
+      via the Foundry package manager [Soldeer]
+      - (Alternative) You can also skip previous steps, copy the contents
+      to your foundry.toml file and run:
+
+   ```shell
+   forge soldeer install
+   ```
+
+3. Install rust toolchain (nightly):
+
+   ```shell
+   curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+   rustup toolchain install nightly
+   rustup default nightly
+   ```
+
+   Note: `@oasisprotocol-sapphire-foundry` relies on Oasis Sapphire
+   `runtime-sdk` crate which requires nightly toolchain.
+
+4. Install Sapphire precompiles:
+   ```shell
+   cd dependencies/@oasisprotocol-sapphire-foundry-0.1.0/precompiles
+   cargo +nightly build --release
+   ```
+
+## Sapphire-specific Tests
+
+Users who aren't familiar with confidential benefits of Sapphire
+should take a moment to review the Ethereum [comparison page].
+The `@oasisprotocol-sapphire-foundry` package enables
+testing with precompiles and encrypted calls.
+
+### Precompiles
+
+Sapphire precompiles are special precompiled contracts that
+contain cryptographic primitives and enable confidential transactions.
+Since Foundry uses un-customized Revm, they are not available by default.
+
+To enable them in Forge tests, add the **import** statement to your test file:
+
+```solidity
+import {SapphireTest} from "@oasisprotocol-sapphire-foundry-0.1.0/BaseSapphireTest.sol";
+```
+
+and then inherit from `SapphireTest` from `BaseSapphireTest.sol`
+and override the `setUp()` function:
+
+```solidity
+   contract PrecompileTest is SapphireTest {
+      
+      function setUp() public override {
+         super.setUp();
+      }
+```
+
+### Encrypted Transactions and Calls
+
+Sapphire uses end-to-end encryption for confidential transactions and calls.
+This means that the calldata is encrypted using the shared [key].
+For non-encrypted transactions, the process works same as on Ethereum.
+However, when testing for example [gasless transactions] with Foundry,
+we need to add a few things.
+
+After deploying the precompiles in the previous step, we need to
+update our custom contract, encrypt the transaction and broadcast it.
+
+1. `SapphireDecryptor` is a special contract that implements decryption
+through the fallback function.
+We need to inherit from it to enable decryption.
+
+```solidity
+   import {SapphireDecryptor} from "@oasisprotocol-sapphire-foundry-0.1.0/BinaryContracts.sol";
+
+   contract CustomContract is SapphireDecryptor {
+```
```

Comment:
````suggestion
   ```solidity
   import {SapphireDecryptor} from "@oasisprotocol-sapphire-foundry-0.1.0/BinaryContracts.sol";

   contract CustomContract is SapphireDecryptor {
   ```
````

Reply:
Updated.